<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1896</width>
    <height>1348</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Silicon Heaven Spy</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <property name="spacing">
     <number>1</number>
    </property>
    <property name="leftMargin">
     <number>1</number>
    </property>
    <property name="topMargin">
     <number>1</number>
    </property>
    <property name="rightMargin">
     <number>1</number>
    </property>
    <property name="bottomMargin">
     <number>1</number>
    </property>
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="leftMargin">
        <number>5</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Silicon Heaven Spy</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1896</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actHelpAbout"/>
   </widget>
   <addaction name="menu_View"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actAddServer"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockServers">
   <property name="allowedAreas">
    <set>Qt::DockWidgetArea::BottomDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Servers</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockServersContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="ServerTreeView" name="treeServers">
       <property name="contextMenuPolicy">
        <enum>Qt::ContextMenuPolicy::CustomContextMenu</enum>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockAttributes">
   <property name="allowedAreas">
    <set>Qt::DockWidgetArea::BottomDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Attributes</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockPropertiesContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Shv path</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edAttributesShvPath"/>
       </item>
       <item>
        <widget class="QPushButton" name="btGotoShvPath">
         <property name="toolTip">
          <string>Go to shv path</string>
         </property>
         <property name="icon">
          <iconset resource="../shvspy.qrc">
           <normaloff>:/shvspy/images/run.png</normaloff>:/shvspy/images/run.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btResizeColumns">
         <property name="toolTip">
          <string>Resize columns to fit</string>
         </property>
         <property name="icon">
          <iconset resource="../shvspy.qrc">
           <normaloff>:/shvspy/images/fit-width.svg</normaloff>:/shvspy/images/fit-width.svg</iconset>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btFileUpload">
         <property name="text">
          <string>Upload</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btFileDownload">
         <property name="text">
          <string>Download</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btLogInspector">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Log inspector</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTableView" name="tblAttributes">
       <property name="editTriggers">
        <set>QAbstractItemView::EditTrigger::AnyKeyPressed|QAbstractItemView::EditTrigger::EditKeyPressed|QAbstractItemView::EditTrigger::SelectedClicked</set>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockNotifications">
   <property name="allowedAreas">
    <set>Qt::DockWidgetArea::BottomDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Notifications</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="shv::visu::LogWidget" name="notificationsLogWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockErrors">
   <property name="windowTitle">
    <string>Log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <property name="spacing">
      <number>3</number>
     </property>
     <property name="leftMargin">
      <number>3</number>
     </property>
     <property name="topMargin">
      <number>3</number>
     </property>
     <property name="rightMargin">
      <number>3</number>
     </property>
     <property name="bottomMargin">
      <number>3</number>
     </property>
     <item>
      <widget class="shv::visu::LogWidget" name="errorLogWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockSubscriptions">
   <property name="windowTitle">
    <string>Subscriptions</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <property name="spacing">
      <number>1</number>
     </property>
     <property name="leftMargin">
      <number>1</number>
     </property>
     <property name="topMargin">
      <number>1</number>
     </property>
     <property name="rightMargin">
      <number>1</number>
     </property>
     <property name="bottomMargin">
      <number>1</number>
     </property>
     <item>
      <widget class="SubscriptionsWidget" name="subscriptionsWidget" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actAddServer">
   <property name="icon">
    <iconset resource="../shvspy.qrc">
     <normaloff>:/shvspy/images/plus.png</normaloff>:/shvspy/images/plus.png</iconset>
   </property>
   <property name="text">
    <string>Add server</string>
   </property>
  </action>
  <action name="actEditServer">
   <property name="icon">
    <iconset resource="../shvspy.qrc">
     <normaloff>:/shvspy/images/edit.png</normaloff>:/shvspy/images/edit.png</iconset>
   </property>
   <property name="text">
    <string>Server Properties</string>
   </property>
  </action>
  <action name="actRemoveServer">
   <property name="icon">
    <iconset resource="../shvspy.qrc">
     <normaloff>:/shvspy/images/minus.png</normaloff>:/shvspy/images/minus.png</iconset>
   </property>
   <property name="text">
    <string>Remove Server</string>
   </property>
  </action>
  <action name="actCopyServer">
   <property name="text">
    <string>Copy server</string>
   </property>
   <property name="toolTip">
    <string>Copy server</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="toolTip">
    <string>Quit ShvSpy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actHelpAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SubscriptionsWidget</class>
   <extends>QWidget</extends>
   <header>src/subscriptionswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ServerTreeView</class>
   <extends>QTreeView</extends>
   <header>src/servertreeview.h</header>
  </customwidget>
  <customwidget>
   <class>shv::visu::LogWidget</class>
   <extends>QWidget</extends>
   <header>shv/visu/logwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../shvspy.qrc"/>
 </resources>
 <connections/>
</ui>
